lib/GHCup/Utils/Pager.hs

```diff
@@ -0,0 +1,54 @@
+{-# LANGUAGE CPP               #-}
+{-# LANGUAGE DataKinds         #-}
+{-# LANGUAGE FlexibleContexts  #-}
+{-# LANGUAGE RankNTypes  #-}
+
+module GHCup.Utils.Pager where
+
+import GHCup.Prelude
+import System.Environment
+import GHCup.Utils.Dirs (findExecutable)
+import Control.Applicative ((<|>))
+import Conduit
+import Data.Conduit.Binary
+import System.Process
+import System.Exit
+import System.IO
+import Data.ByteString (ByteString)
+import Data.Text (Text)
+
+import qualified Data.ByteString.Lazy as BSL
+import qualified Data.Text.Encoding   as E
+
+getPager :: IO (Maybe FilePath)
+getPager
+  | isWindows = pure Nothing
+  | otherwise = do
+      lookupEnv "GHCUP_PAGER" >>= \case
+        Just r  -> pure $ Just r
+        Nothing -> lookupEnv "PAGER" >>= \case
+          Just r' -> pure $ Just r'
+          Nothing -> do
+                findExecutable "most"
+            <|> findExecutable "more"
+            <|> findExecutable "less"
+
+sendToPager :: FilePath -> Text -> IO ()
+sendToPager pager text = sendToPagerConduit pager (sourceLbs (BSL.fromStrict . E.encodeUtf8 $ text))
```

Comment:
Assuming UTF-8 here is a bit unfortunate. But I don't know if ghcup behaves well wrt unicode symbols if you have esoteric encodings set anyway.